### DISCUSSION TITLE
Java minimum version for source files: upgrade?

### DISCUSSION THREAD

**Discussion by Jiehong** [remkop/picocli#1913] [General]
Hello there,

In the CONTRIBUTING.md file, we can see the following table:

Artifact | Target Java Version
------------ | -------------
`picocli` | Java 5
`picocli-codegen` | Java 6
`picocli-groovy` | Java 5
`picocli-shell-jline2` | Java 5
`picocli-shell-jline3` | Java 8
`picocli-spring-boot-starter` | Java 8

What's the reason for keeping a java 5 language version compatibility when no vendor supports that version anymore?

According to https://en.wikipedia.org/wiki/Java_version_history#Release_table, we could at least raise it to java 6 if we consider Azul, or 8 otherwise.

WDYT?

**Comment by remkop** [comment]
I'm not thinking about this in terms of vendor support. 
I'm thinking how can I provide maximum value to the maximum amount of people. 

There may be people who are, for some reason, stuck on Java 5. What is the drawback of supporting that use case? How painful is it to ensure that picocli can be used on Java 5? 

And the answer so far is that there's nothing that I want to do that I can't do on Java 5. So why not support Java 5?

If it becomes too painful I'll reconsider but so far I see no reason to require Java 6 or 8 or whatever...

**Reply by Jiehong** [reply]
Sure! Having a clear reason is enough to me.
